src/index.ts

```diff
@@ -5,22 +5,20 @@ import { DependencyList, EffectCallback, useEffect, useLayoutEffect, useRef } fr
  * @see https://www.npmjs.com/package/eslint-plugin-react-hooks#advanced-configuration
  */
 
-const depsFallback: DependencyList = [];
```

Comment:
The reason the dep fallback is out of the hook is to make sure the reference remains the same across rerenders - so it's actually intentionally :)